cmd/cmd.go

```diff
@@ -385,28 +389,39 @@ func StartNode(workingDir string, passwordFetcher func(*wallet.Wallet) (string,
 	}
 
 	// Create reward addresses
-	rewardAddrs := make([]crypto.Address, 0, len(allValAddrs))
+	rewardAddrs := make([]crypto.Address, 0, len(valAddrsInfo))
 	if len(conf.Node.RewardAddresses) != 0 {
 		for _, addrStr := range conf.Node.RewardAddresses {
 			addr, _ := crypto.AddressFromString(addrStr)
 			rewardAddrs = append(rewardAddrs, addr)
 		}
 	} else {
-		for i := 0; i < len(allValAddrs); i++ {
-			valAddrPath, _ := addresspath.NewPathFromString(allValAddrs[i].Path)
-			accAddrPath := addresspath.NewPath(valAddrPath.Purpose(), valAddrPath.CoinType(),
-				uint32(crypto.AddressTypeValidator)+hdkeychain.HardenedKeyStart, valAddrPath.AddressIndex())
+		for i := 0; i < len(valAddrsInfo); i++ {
+			valAddrPath, _ := addresspath.NewPathFromString(valAddrsInfo[i].Path)
+			accAddrPath := addresspath.NewPath(
+				valAddrPath.Purpose(),
+				valAddrPath.CoinType(),
+				uint32(crypto.AddressTypeBLSAccount)+hdkeychain.HardenedKeyStart,
```

Comment:
@amirvalhalla 
Reward address can't be a validator address, otherwise the block will be invalid.
I added a check to cover this.